Encapsulate the Spotlight config

jcoyne opened the pull request:
All values set on Spotlight::Engine.config are actually stored
in a class variable on Railtie::Configuration. So these pollute
that namespace.  All other railties are encapsulating their configs
onto a single entry in that scope.  This change makes Spotlight do
likewise.

This probably needs to be a major revision

hackartisan reviewed (commented):
We looked at this in a PR review ensemble. We're interested in whether this could be made more backwards compatible.

- could we check first for the spotlight namespaced config, then fall back if it's not found?
- could we use an initializer to pull the configs we expect to find into a spotlight namespace?
- other ideas?

jcoyne committed:
Encapsulate the Spotlight config

All values set on Spotlight::Engine.config are actually stored
in a class variable on Railtie::Configuration. So these pollute
that namespace. All other railties are encapsulating their configs
onto a single entry in that scope. This change makes Spotlight do
likewise.

This probably needs to be a major revision
